### PR TITLE
feat(vscode): Expose tree view

### PR DESCRIPTION
**Pull request by mrmeku**

### CODE REVIEW

File: apps/vscode/src/app/tree-view/current-workspace-tree-provider.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  static create(
+    workspacePath: string | undefined,
+    extensionPath: string
+  ): TreeDataProvider<WorkspaceRoute> {
+    return new CurrentWorkspaceTreeProvider(
+      workspacePath && existsSync(join(workspacePath, 'angular.json'))
+        ? {
+            path: workspacePath,
+            name: JSON.parse(
+              readFileSync(join(workspacePath, 'angular.json'), 'utf8')
```
Comment (vsavkin) [inline apps/vscode/src/app/tree-view/current-workspace-tree-provider.ts]: add a catch around it, so if json is not parseable, the extension still shows something

even better, maybe try to parse out the name without parsing json

File: apps/angular-console/src/app/app.component.ts [hunk 2]
```diff
@@ -50,10 +50,19 @@ export class AppComponent implements OnInit, OnDestroy {
...
+    router: Router
   ) {
     settings.fetch().subscribe();
+
+    router.events
+      .pipe(filter((e): e is NavigationEnd => e instanceof NavigationEnd))
+      .subscribe(e => {
+        if (e.urlAfterRedirects === '/connect') {
+          this.contextualActionBarService.breadcrumbs$.next([
+            { title: 'Connect' }
```
Comment (vsavkin) [inline apps/angular-console/src/app/app.component.ts]: why is this needed?

File: package.json [hunk 3]
```diff
@@ -76,14 +76,33 @@
     ]
   },
   "dependencies": {
-    "@nrwl/nx": "7.4.0"
+    "@angular/cli": "7.2.3",
```
[the remark this answers is not in the document]
Comment (mrmeku) [reply package.json]: Good catch. I've updated our dependencies to mirror our actual run-time dependencies now (we only have 3 now which is great)

File: tools/scripts/vscode-npm.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const shell = require('shelljs');
+
+shell.exec(
+  'npm install --prod --no-optional --no-shrinkwrap --no-audit --no-bin-link',
```
[the remark this answers is not in the document]
Comment (mrmeku) [reply tools/scripts/vscode-npm.js]: Before Jason helped me out it shrank the .vsix file by around 10 megs using npm. Now that we only have 3 runtime deps the bundle sizes are now equivalent. I'll switch back

File: libs/server/src/lib/api/read-settings.ts [hunk 5]
```diff
@@ -31,6 +33,10 @@ export function readSettings(store: Store): Settings {
   return settings;
 }
 
+const storeSettingsSubject = new Subject<any>();
```
[the remark this answers is not in the document]
Comment (mrmeku) [reply libs/server/src/lib/api/read-settings.ts]: Our VSCode extension doesn't have a DI mechanism right now so for now this is the best we can do. We can revisit adding DI to VScode later on

File: libs/feature-workspaces/src/lib/feature-workspaces.module.ts [hunk 6]
```diff
@@ -43,6 +43,10 @@ export const workspaceRoutes: Route[] = [
     component: NewWorkspaceComponent,
     data: { state: CREATE_WORKSPACE }
   },
+  {
```
[the remark this answers is not in the document]
Comment (mrmeku) [reply libs/feature-workspaces/src/lib/feature-workspaces.module.ts]: We do it twice so that we can do /connect or /workspace/WORKSPACE-PATH/connect.

Until we make some alternations to the electron version of the app this is necessary

File: apps/vscode/src/app/tree-view/current-workspace-tree-provider.ts (hunk 1 above)
Comment (mrmeku) [reply apps/vscode/src/app/tree-view/current-workspace-tree-provider.ts]: I added a try catch block. I'm not confident enough in my regexp to make a reliable way to strip the name without parsing.

File: apps/angular-console/src/app/app.component.ts (hunk 2 above)
Comment (mrmeku) [reply apps/angular-console/src/app/app.component.ts]: Its needed only because the narwhal extension do not set their own breadcrumbs. I'll mark it with a todo to move the logic over to the other repo

File: libs/server/src/lib/server.module.ts [hunk 7]
```diff
@@ -1,21 +1,22 @@
...
 import { AngularConsoleExtensionsModule } from '@nrwl/angular-console-enterprise-electron';
+import * as path from 'path';
+
+import { docs } from './api/docs';
 import { readSettings } from './api/read-settings';
 import { commands } from './api/run-command';
-import { Telemetry } from './utils/telemetry';
-import { docs } from './api/docs';
 import { DefaultController } from './default.controller';
+import { ArchitectResolver } from './resolvers/architect.resolver';
```
[the remark this answers is not in the document]
Comment (mrmeku) [reply libs/server/src/lib/server.module.ts]: Donezo. I'll turn off the automatic sorting extension I have

File: libs/feature-workspaces/src/lib/feature-workspaces.module.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (vsavkin) [reply libs/feature-workspaces/src/lib/feature-workspaces.module.ts]: Can we instead make it such that we always go to /connect. Something like this:

```
     if (this.settings.showConnectPlugin()) {
        this.routes.splice(this.routes.length - 2, 0, {
          icon: 'question_answer',
          url: '/connect',
          title: 'Connect'
        });
      }
```

The connect route should be nested in a workspace. We don't use the workspace information in any way right now.
